src/service/logs/kinesis_firehose.rs

```diff
@@ -0,0 +1,368 @@
+use crate::meta::StreamType;
+use crate::service::db;
+
+use super::StreamMeta;
+
+pub async fn process(
+    org_id: &str,
+    stream_name: &str,
+    request: KinesisFHRequest,
+    thread_id: web::Data<usize>,
+) -> Result<HttpResponse, Error> {
+    let start = Instant::now();
+
+    if !cluster::is_ingester(&cluster::LOCAL_NODE_ROLE) {
+        return Ok(
+            HttpResponse::InternalServerError().json(MetaHttpResponse::error(
+                http::StatusCode::INTERNAL_SERVER_ERROR.into(),
+                "not an ingester".to_string(),
+            )),
+        );
+    }
+
+    // check if we are allowed to ingest
+    if db::compact::delete::is_deleting_stream(org_id, stream_name, StreamType::Logs, None) {
+        return Ok(
+            HttpResponse::InternalServerError().json(MetaHttpResponse::error(
+                http::StatusCode::INTERNAL_SERVER_ERROR.into(),
+                format!("stream [{stream_name}] is being deleted"),
+            )),
+        );
+    }
+
+    #[cfg(feature = "zo_functions")]
+    let lua = Lua::new();
+    #[cfg(feature = "zo_functions")]
+    let state = vrl::state::Runtime::default();
+    #[cfg(feature = "zo_functions")]
+    let mut runtime = vrl::Runtime::new(state);
+
+    let mut min_ts =
+        (Utc::now() + Duration::hours(CONFIG.limit.ingest_allowed_upto)).timestamp_micros();
+
+    let mut stream_schema_map: AHashMap<String, Schema> = AHashMap::new();
+    let mut stream_alerts_map: AHashMap<String, Vec<Alert>> = AHashMap::new();
+    let mut stream_status = StreamStatus {
+        name: stream_name.to_owned(),
+        status: RecordStatus {
+            successful: 0,
+            failed: 0,
+            error: "".to_string(),
+        },
+    };
+    let mut trigger: Option<Trigger> = None;
+
+    // Start Register Transforms for stream
+    #[cfg(feature = "zo_functions")]
+    let (local_tans, stream_lua_map, stream_vrl_map) =
+        crate::service::ingestion::register_stream_transforms(
+            org_id,
+            stream_name,
+            StreamType::Logs,
+            &lua,
+        );
+    // End Register Transforms for stream
+
+    let stream_schema = crate::service::schema::stream_schema_exists(
+        org_id,
+        stream_name,
+        StreamType::Logs,
+        &mut stream_schema_map,
+    )
+    .await;
+    let mut partition_keys: Vec<String> = vec![];
+    if stream_schema.has_partition_keys {
+        partition_keys = crate::service::ingestion::get_stream_partition_keys(
+            stream_name.to_string(),
+            stream_schema_map.clone(),
+        )
+        .await;
+    }
+    // Start get stream alerts
+    let key = format!("{}/{}/{}", &org_id, StreamType::Logs, &stream_name);
+    crate::service::ingestion::get_stream_alerts(key, &mut stream_alerts_map).await;
+    // End get stream alert
+
+    let mut buf: AHashMap<String, Vec<String>> = AHashMap::new();
+    for (_i, record) in request.records.iter().enumerate() {
```

Comment:
You don't use `_i`, so there's no need to enumerate.
```suggestion
    for record in request.records {
```